fastddsspy_participants/test/unittest/visualization/ModelParserTest.cpp

```diff
@@ -0,0 +1,661 @@
+    ASSERT_EQ(result.guid, expected_result.guid);
+    ASSERT_EQ(result.name, expected_result.name);
+    unsigned int i = 0;
+    for (const auto& writer : result.writers)
+    {
+        ASSERT_EQ(writer.topic_name, expected_result.writers[i].topic_name);
+        ASSERT_EQ(writer.topic_type, expected_result.writers[i].topic_type);
+        ASSERT_EQ(writer.number, expected_result.writers[i].number);
+        i++;
+    }
+    i = 0;
+    for (const auto& reader : result.readers)
+    {
+        ASSERT_EQ(reader.topic_name, expected_result.readers[i].topic_name);
+        ASSERT_EQ(reader.topic_type, expected_result.readers[i].topic_type);
+        ASSERT_EQ(reader.number, expected_result.readers[i].number);
+        i++;
+    }
+    ASSERT_EQ(model.participant_database_.size(), 1);
+    ASSERT_EQ(model.endpoint_database_.size(), 2);
+
+}
+
+TEST(ModelParserTest, simple_endpoint_writer)
+{
+    // Create model
+    spy::participants::SpyModel model;
+    // Fill model
+    spy::participants::EndpointInfo endpoint;
+    random_endpoint_info(endpoint, ddspipe::core::types::EndpointKind::writer, true, 2);
+    model.endpoint_database_.add(endpoint.guid, endpoint);
+
+    // Obtain information from model
+    std::vector<spy::participants::SimpleEndpointData> result;
+    result = spy::participants::ModelParser::writers(model);
+
+    // Create expected return
+    std::vector<spy::participants::SimpleEndpointData> expected_result;
+    expected_result.push_back({
+        endpoint.guid,
+        endpoint.discoverer_participant_id,
+        {
+            endpoint.topic.m_topic_name,
+            endpoint.topic.type_name
+        }
+
+    });
+
+    // Check information
+    ASSERT_EQ(result[0].guid, expected_result[0].guid);
+    ASSERT_EQ(result[0].topic.topic_name, expected_result[0].topic.topic_name);
+    ASSERT_EQ(result[0].topic.topic_type, expected_result[0].topic.topic_type);
+}
+
+TEST(ModelParserTest, simple_endpoint_reader)
+{
+    // Create model
+    spy::participants::SpyModel model;
+    // Fill model
+    spy::participants::EndpointInfo endpoint;
+    random_endpoint_info(endpoint, ddspipe::core::types::EndpointKind::reader, true, 2);
+    model.endpoint_database_.add(endpoint.guid, endpoint);
+
+    // Obtain information from model
+    std::vector<spy::participants::SimpleEndpointData> result;
+    result = spy::participants::ModelParser::readers(model);
+
+    // Create expected return
+    std::vector<spy::participants::SimpleEndpointData> expected_result;
+    expected_result.push_back({
+        endpoint.guid,
+        endpoint.discoverer_participant_id,
+        {
+            endpoint.topic.m_topic_name,
+            endpoint.topic.type_name
+        }
+    });
+
+    // Check information
+    ASSERT_EQ(result[0].guid, expected_result[0].guid);
+    ASSERT_EQ(result[0].topic.topic_name, expected_result[0].topic.topic_name);
+    ASSERT_EQ(result[0].topic.topic_type, expected_result[0].topic.topic_type);
+}
+
```

Comment:
Please, add comments to each test. This test name is not enough to understand what the test does